File: IM-APP-server/internal/models/file.go
package models

// 文件模块 DTO（按 GOAL-APP 清单）

type FileObject struct {
	ID           string `json:"id"`
	Purpose      string `json:"purpose"` // avatar|image|voice|file|sticker
	FileName     string `json:"fileName"`
	ContentType  string `json:"contentType"`
	Size         int64  `json:"size"`
	URL          string `json:"url"`
	ThumbnailURL string `json:"thumbnailUrl,omitempty"`
	DurationMs   int64  `json:"durationMs,omitempty"`
	Status       string `json:"status"` // pending|ready|rejected
}

type CreateUploadRequest struct {
	Purpose     string `json:"purpose"` // avatar|image|voice|file|sticker
	FileName    string `json:"fileName"`
	ContentType string `json:"contentType"`
	Size        int64  `json:"size"`
	SHA256      string `json:"sha256,omitempty"`
}

type UploadInitResult struct {
	File      FileObject        `json:"file"`
	UploadURL string            `json:"uploadUrl"`
	FormURL   string            `json:"formUrl,omitempty"`
	FormData  map[string]string `json:"formData,omitempty"`
	Headers   map[string]string `json:"headers,omitempty"`
	ExpiresIn int               `json:"expiresIn"`
}

type CompleteUploadRequest struct {
	FileID string `json:"fileId"`
	ETag   string `json:"etag,omitempty"`
}

File: IM-APP-fronend/src/types/file.ts
/** 文件上传相关类型（对齐 Apifox） */

export type UploadPurpose = 'avatar' | 'image' | 'voice' | 'file' | 'sticker'

export interface CreateUploadInput {
  purpose: UploadPurpose
  fileName: string
  contentType: string
  size: number
  sha256?: string
}

export interface FileObject {
  id: string
  purpose?: string
  fileName?: string
  contentType?: string
  size?: number
  url?: string
  thumbnailUrl?: string
  durationMs?: number
  status?: string
}

export interface UploadInitResult {
  file: FileObject
  uploadUrl: string
  formUrl?: string
  formData?: Record<string, string>
  headers?: Record<string, string>
  expiresIn: number
}

export interface CompleteUploadInput {
  etag?: string
}

/** @deprecated 使用 FileObject */
export type FileInfo = FileObject

/** @deprecated 旧版 presign 响应 */
export interface PresignResult {
  uploadUrl: string
  fileUrl: string
  objectKey: string
  expiresIn: number
  devMode?: boolean
}
